ch_backup/storage/async_pipeline/pipeline_executor.py

```diff
@@ -169,15 +170,30 @@ def wait(self, keep_going: bool = False) -> None:
         if self._exec_pool:
             self._exec_pool.wait_all(keep_going)
 
-    def _exec_pipeline(self, job_id: str, pipeline: Callable, is_async: bool) -> Any:
+    def _exec_pipeline(
+        self,
+        job_id: str,
+        pipeline: Callable,
+        is_async: bool,
+        callback: Optional[Callable] = None,
+    ) -> Any:
         """
         Run pipeline inplace or schedule for exec in process pool
         """
 
         if is_async and self._exec_pool:
-            return self._exec_pool.submit(job_id, pipeline)
-
-        return pipeline()
+            return self._exec_pool.submit(job_id, pipeline, callback)
+
+        try:
+            result = pipeline()
+        except Exception as ex:
+            if callback:
```

Comment:
When exception occurs then we will call a `callback(ex)`
But inside `class UploadPartObserver` we will just ignore it. Why? Maybe remove it?